great_expectations/data_context/data_context/abstract_data_context.py

```diff
@@ -1577,10 +1593,62 @@ def get_validator(
         include_rendered_content: Optional[bool] = None,
         **kwargs: Optional[dict],
     ) -> Validator:
-        """
-        This method applies only to the new (V3) Datasource schema.
-        """
+        """Retrieves a validator with a Batch list and an Expectation Suite.
 
+        `get_validator` first calls `get_batch_list` to retrieve a batch list, then creates or retrieves
+        an ExpectationSuite used to validate the Batches in the list.
+
+        Args:
+            datasource_name: The name of the Datasource that defines the Data Asset to retrieve the batch for
+            data_connector_name: The Data Connector within the datasource for the Data Asset
+            data_asset_name: The name of the Data Asset within the Data Connector
+            batch: The Batch to use with the Validator
+            batch_list: The List of Batches to use with the Validator
+            batch_request: Encapsulates all the parameters used here to retrieve a BatchList. Use either
+                `batch_request` or the other params (but not both)
+            batch_request_list: A List of BatchRequest to use with the Validator
+            batch_data: Provides runtime data for the batch; is added as the key `batch_data` to
+                the `runtime_parameters` dictionary of a BatchRequest
+            query: Provides runtime data for the batch; is added as the key `query` to
+                the `runtime_parameters` dictionary of a BatchRequest
+            path: Provides runtime data for the batch; is added as the key `path` to
+                the `runtime_parameters` dictionary of a BatchRequest
+            runtime_parameters: Specifies runtime parameters for the BatchRequest; can includes keys `batch_data`,
+                `query`, and `path`
+            data_connector_query: Used to specify connector query parameters; specifically `batch_filter_parameters`,
+                `limit`, `index`, and `custom_filter_function`
+            batch_identifiers: Any identifiers of batches for the BatchRequest
+            batch_filter_parameters: Filter parameters used in the data connector query
+            limit: Part of the data_connector_query, limits the number of rows in a batch
```

Comment:
```suggestion
            limit: Part of the data_connector_query, limits the number of batches in the batch list
```

Reply:
Verified with the following tweaks to `test_get_validator()`. The first assertion len(batches) == 2 passes. The other assertions on rows in batches fail. We can consider adding a test if this isn't already tested:
```
def test_get_validator(data_context_with_simple_sql_datasource_for_testing_get_batch):
    context = data_context_with_simple_sql_datasource_for_testing_get_batch
    context.create_expectation_suite("my_expectations")

    # A BatchRequest specified without the date batch_filter_parameter will return all 30 batches.
    validator = context.get_validator(
        datasource_name="my_sqlite_db",
        data_connector_name="daily",
        data_asset_name="table_partitioned_by_date_column__A",
        limit=2,
        expectation_suite_name="my_expectations",
    )

    batches = validator.batches

    assert len(batches) == 2
    # assert len(batches[list(batches.keys())[0]].head(100)) == 2
    row_count = validator.get_metric(
        MetricConfiguration(
            "table.row_count",
            metric_domain_kwargs={"batch_id": validator.active_batch_id},
        )
    )
    batch_shapes = []
    for id, batch in batches.items():
        batch_shapes.append(batch.head(100).shape)
    for id, batch in batches.items():
        num_rows = batch.head(100).shape[0]
        assert num_rows <= 2
    df = batches[validator.active_batch_id]
    assert row_count == 2
```